secretmanager/quickstart.rb

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def quickstart i_project_id: nil, i_secret_id: nil
+  # [START secretmanager_quickstart]
+  # Import the Secret Manager client library.
+  require "google/cloud/secret_manager"
+
+  # GCP project in which to store secrets in Secret Manager.
+  project_id = "YOUR_PROJECT_ID"
+
+  # ID of the secret to create.
+  secret_id = "YOUR_SECRET_ID"
+
+  # [END secretmanager_quickstart]
+  project_id = i_project_id if i_project_id
+  secret_id = i_secret_id if i_secret_id
+  # [START secretmanager_quickstart]
+  # Create the Secret Manager client.
+  client = Google::Cloud::SecretManager.secret_manager_service
+
+  # Build the parent name from the project.
+  parent = "projects/#{project_id}"
+
+  # Create the parent secret.
+  secret = client.create_secret(
+    parent:    parent,
+    secret_id: secret_id,
+    secret:    {
+      replication: {
+        automatic: Google::Cloud::SecretManager::V1beta1::Replication::Automatic.new
```

Comment:
`automatic: {}` will work here as well, and may be less annoying. (Same for the "create_secret" snippet.)